### PR TITLE
feat: Added Categories

### LINKED ISSUE #89: Need categories on top of the landing page
Categories are going to be
1. Web Development (Basics)
2. Web Development (Advance)
3. Devops
4. Web3

Screenshot 2024-03-20 at 3 29 52 PM

### PR DESCRIPTION
**Pull request by ShubhamMewara**
Fixed #89 and #123 

https://github.com/code100x/daily-code/assets/136337515/9e256667-6d13-4acc-a0d1-497985f666f5

### CODE REVIEW

File: packages/db/prisma/schema.prisma [hunk 1]
```diff
@@ -54,6 +55,13 @@ enum ProblemType {
   Code
   MCQ
 }
+enum Category {
+  Fundamentals
```
Comment (hkirat) [inline packages/db/prisma/schema.prisma:59]: There should be a saparate table called categories
Each track should have one to many to it. Single track can be part of two categories
Everything else looks good

**Review by hkirat — CHANGES_REQUESTED** [review]

**Comment by ShubhamMewara** [comment]
hey @hkirat 
Please let me know if you need any further adjustments or have additional tasks.

https://github.com/code100x/daily-code/assets/136337515/a2b13337-d6b0-4990-b613-e7c315f16719

**Comment by krishna9358** [comment]
@ShubhamMewara One suggestion, instead of making a row of categories try to make a dropdown button if possible. UI will look better imo.

File: packages/ui/src/Categories.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+export const Categories = () => {
+  function handleFilterButton(category: string) {
+    if (category == selectedCategory) {
+      setSelectedCategory("");
+    } else {
+      setSelectedCategory(category);
+    }
+  }
+  const categories = ["Fundamentals", "Advance", "Devops", "Web3", "DSA"];
```
Comment (hkirat) [inline packages/ui/src/Categories.tsx:16]: Should come from the DB

**Review by hkirat — CHANGES_REQUESTED** [review]
merge conflicts, and also all categories should come from the DB

File: packages/db/prisma/schema.prisma (hunk 1 above)
Comment (ShubhamMewara) [reply packages/db/prisma/schema.prisma]: @hkirat 
I have made changes in #148 can you please review it?

File: packages/ui/src/Categories.tsx (hunk 2 above)
Comment (ShubhamMewara) [reply packages/ui/src/Categories.tsx]: @hkirat 
I have made changes in #148 can you please review it?

**Pull request closed without merging**